java/java.hints/src/org/netbeans/modules/java/hints/introduce/Flow.java

```diff
@@ -444,11 +445,45 @@ protected boolean isCanceled() {
             return cancel.isCanceled();
         }
 
+        private TreePath currentPath;
+
+        public TreePath getCurrentPath() {
+            return currentPath;
+        }
+
+        public Boolean scan(TreePath path, ConstructorData p) {
+            TreePath oldPath = currentPath;
+            try {
+                currentPath = path;
+                return super.scan(path.getLeaf(), p);
+            } finally {
+                currentPath = oldPath;
+            }
+        }
+
         @Override
         public Boolean scan(Tree tree, ConstructorData p) {
             resume(tree, resumeBefore);
             
-            Boolean result = super.scan(tree, p);
+            Boolean result;
+
+            if (tree != null) {
```

Comment:
Wouldn't it be beneficial to build the `visitSwitchExpression` and `visitYield` into baseclasses ?